### PR TITLE
Acceleration

### PR DESCRIPTION
**Pull request by TobiasWinchen**
Implementation of first and second order Fermi acceleration.

The code is documented in the new notebooks in the doc directory:
https://github.com/TobiasWinchen/CRPropa3/blob/acceleration/doc/pages/example_notebooks/acceleration

While integrating the examples I slightly restructured and streamlined the existing documentation.

### CODE REVIEW

File: include/crpropa/module/Acceleration.h
Comment (adundovi) [inline include/crpropa/module/Acceleration.h:13]: Do we have brace placement style agreed? Is it formatted with `clang-format`?

File: doc/pages/acceleration.rst
Comment (rafaelab) [inline doc/pages/acceleration.rst:8]: The scattering process -> In the scattering process,

File: doc/pages/acceleration.rst
Comment (rafaelab) [inline doc/pages/acceleration.rst:6]: scatter center -> scattering center

File: doc/pages/acceleration.rst
Comment (rafaelab) [inline doc/pages/acceleration.rst:9]: scatter center -> scattering center

Comment (rafaelab) [reply doc/pages/acceleration.rst:9]: Also in other places below

File: include/crpropa/module/Acceleration.h
Comment (rafaelab) [inline include/crpropa/module/Acceleration.h:1]: To keep a consistent convention, should be CRPROPA_ACCELERATION.

File: include/crpropa/module/Acceleration.h
Comment (rafaelab) [inline include/crpropa/module/Acceleration.h:125]: For style convention, since we can use kwargs in python, shouldn't it be turbulenceIndex in camel case?

File: include/crpropa/module/Acceleration.h
Comment (TobiasWinchen) [reply include/crpropa/module/Acceleration.h]: I updated the formatting

File: include/crpropa/module/Acceleration.h
Comment (rafaelab) [inline include/crpropa/module/Acceleration.h]: Same comment as before: if we can interact with a variable and use it as keyword in python, shouldn't we stick to our convention (camel case)?

**Comment by TobiasWinchen** [comment]
Thanks for your comments, I updated the spelling + coding style accordingly.

**Comment by adundovi** [comment]
Hi Tobi, it is OK from my side. However, I checked it on a technical level only, not equations and physics, yet I assume you test this carefully.

**Review by adundovi — APPROVED** [review]

**Comment by rafaelab** [comment]
I am still testing this. I will merge the PR once I'm done (probably early next week).
I'm just concerned that some combinations of energy losses and energy gains might mess things up, so I'm testing some limiting cases.  I haven't found anything so far. 
Btw, great work, @TobiasWinchen !

**Comment by rafaelab** [comment]
@TobiasWinchen I have finished reviewing the PR and testing the module. I have no more comments.
Let's just wait until 3.1.7 is released on Friday, and then we can merge your PR, since this is a feature of 3.2.

**Comment by rafaelab** [comment]
Ready to be merged.
@TobiasWinchen : just make sure to add the relevant information to the Change Log before merging.

**Pull request merged**
